lib/sessions.js

```diff
@@ -0,0 +1,68 @@
+'use strict';
+
+const redis = require('redis');
+const session = require('express-session');
+const connectRedis = require('connect-redis');
+const cookieParser = require('cookie-parser');
+
+module.exports = (app, config) => {
+
+  const encryption = require('./encryption')(config.session.secret);
+
+  const RedisStore = connectRedis(session);
+  const client = redis.createClient(config.redis.port, config.redis.host);
+
+  client.on('error', (e) => {
+    throw e;
+  });
+
+  const store = new RedisStore({
+    client: client,
+    ttl: config.session.ttl,
+    secret: config.session.secret,
+    serializer: {
+      parse: (data) => JSON.parse(encryption.decrypt(data)),
+      stringify: (data) => encryption.encrypt(JSON.stringify(data))
+    }
+  });
+
+  let secure;
+
+  app.use((req, res, next) => {
+    secure = req.protocol === 'https' ? true : false;
+    next();
+  })
+
+  if (secure) {
+    app.set('trust proxy', 1);
+  }
+
+  const sessionConfig = {
+    cookie: {
+      secure: secure
+    },
+    key: config.session.key,
+    secret: config.session.secret,
+    resave: typeof config.session.resave === 'boolean' ? config.session.resave : true,
+    saveUninitialized: typeof config.session.saveUninitialized === 'boolean' ? config.session.saveUninitialized : true
+  }
+
+  if (store) {
+    sessionConfig.store = store;
+  }
+
+  app.use(cookieParser(config.session.secret));
+  app.use((req, res, next) => {
+    const cookie = res.cookie.bind(res);
+    res.cookie = (name, value, options) => {
+      options = options || {};
+      options.secure = secure;
+      options.httpOnly = true;
+      options.path = '/';
+      cookie(name, value, options);
```

Comment:
all the options can be specified as a second argument in the `cookieParser` method. there's no need to rewrap and rebind `res.cookie`.

```javascript
app.use(cookieParser(config.session.secret, {secure: true, httpOnly: true}))
```